packages/components/src/VirtualizedList/utils/resizable.js

```diff
@@ -0,0 +1,267 @@
+import flow from 'lodash/flow';
+
+const MINIMUM_COLUMN_WIDTH = 40;
+
+const getMinimumWidth = column => column.minWidth || MINIMUM_COLUMN_WIDTH;
+
+const getWidth = (width, minWidth = MINIMUM_COLUMN_WIDTH) => (width <= minWidth ? minWidth : width);
+
+/**
+ * Search for the nearest index shrinkable on the right.
+ * @param {integer} index
+ * @param {array} columnsWidths
+ */
+const getShrinkIndexRight = (index, columnsWidths) => {
+	for (let i = index + 1; i < columnsWidths.length; i += 1) {
+		const next = i + 1;
+		// Shrink columns on the right of the current, if resizable and width above minimal.
+		if (columnsWidths[i].resizable && columnsWidths[i].width > getMinimumWidth(columnsWidths[i])) {
+			return i;
+			// Shrink the after first columns on the right, if the next one is already at minimal width.
+		} else if (
+			columnsWidths[i] === getMinimumWidth(columnsWidths[i]) &&
+			next < columnsWidths.length - 1 &&
+			columnsWidths[next].resizable &&
+			columnsWidths[next].width > getMinimumWidth(columnsWidths[next])
+		) {
+			return next;
+		}
+	}
+	return -1;
+};
+
+/**
+ * Search for the nearest index enlargeable on the right.
+ * @param {integer} index
+ * @param {array} columnsWidths
+ */
+const getEnlargeIndexRight = (index, columnsWidths) => {
+	for (let i = index + 1; i < columnsWidths.length; i += 1) {
+		// Enlarge the nearest right column.
+		if (columnsWidths[i].resizable && columnsWidths[i].width > getMinimumWidth(columnsWidths[i])) {
+			return i;
+		}
+		// Enlarge the last column only if trigger by the before last column.
+		if (
+			index === columnsWidths.length - 2 &&
+			i === columnsWidths.length - 1 &&
+			columnsWidths[i].resizable
+		) {
+			return i;
+		}
+	}
+	return -1;
+};
+
+/**
+ * Search for the nearest index shrinkable on the left.
+ * @param {integer} index
+ * @param {array} columnsWidths
+ */
+const getShrinkIndexLeft = (index, columnsWidths) => {
+	for (let i = index; i >= 0; i -= 1) {
+		const previous = i - 1;
+		// Shrink column at left.
+		if (columnsWidths[i].resizable && columnsWidths[i].width > getMinimumWidth(columnsWidths[i])) {
+			return i;
+			// Shrink the after first colum on the left,
+			// if the previous one is already at minimal width.
+		} else if (
+			columnsWidths[i] === getMinimumWidth(columnsWidths[i]) &&
+			previous < 0 &&
+			columnsWidths[previous].resizable &&
+			columnsWidths[previous].width > getMinimumWidth(columnsWidths[previous])
+		) {
+			return previous;
+		}
+	}
+	return -1;
+};
+
+/**
+ * Set the collection with the value, and return it.
+ * @param {boolean} resized the value to set.
+ */
+const setResized = resized => column => ({ ...column, resized });
```

Comment:
I don't get the `resized` attribute.
AFAIU, it is set everytime a column is resized. But it is only used to see if it's a fixed column. But a Fixed column is defined by definition by `resizable`.

In `isFixedColumnWidth` test: `!resizable || !resized || width <= minWidth;` seems weird. It is used to set the flex -shrink and flex-grow. But we should have
* !resizable ==> shrink = grow = 0
* resizable && (width <= minWidth) ==> grow = 1, shrink = 0
* resizable ==> shrink = grow = 1

So what about this `resized` attribute ?

Reply:
Resized (badly named, maybe resizing will be better), indicates the two columns which are currently resizing.

Reply:
In fact my question is why do you need this info ?
Moreover, that's not totally true, you reset it when you start to resize. This is not reset when you stop resizing.